packages/builder/src/utils/template.ts

```diff
@@ -1,8 +1,141 @@
-import Fuse from 'fuse.js';
+import 'ses';
 import _ from 'lodash';
-
+import Debug from 'debug';
+import Fuse from 'fuse.js';
 import type { TemplateOptions } from 'lodash';
+import * as acorn from 'acorn';
+import { Node, Identifier, MemberExpression } from 'acorn';
+import { CannonHelperContext } from '../types';
+
+const debug = Debug('cannon:builder:template');
+
+/**
+ * Gets all global properties from various JavaScript environments
+ * @returns Set of all discovered global properties
+ */
+function getAllGlobalProperties(): Set<string> {
+  const globals = new Set<string>();
+
+  // Helper to safely get all properties from an object and its prototype chain
+  const getAllPropertiesFromObject = (obj: any) => {
+    while (obj) {
+      try {
+        Object.getOwnPropertyNames(obj).forEach((prop) => globals.add(prop));
+        obj = Object.getPrototypeOf(obj);
+      } catch (e) {
+        break; // Stop if we can't access further
+      }
+    }
+  };
+
+  // Check globalThis
+  getAllPropertiesFromObject(globalThis);
+
+  // Check window in browser environment
+  if (typeof window !== 'undefined') {
+    getAllPropertiesFromObject(window);
+  }
+
+  // Check global in Node.js environment
+  if (typeof global !== 'undefined') {
+    getAllPropertiesFromObject(global);
+  }
+
+  // Check self (used in Web Workers)
+  if (typeof self !== 'undefined') {
+    getAllPropertiesFromObject(self);
+  }
+
+  // Get properties from Function constructor
+  try {
+    const functionProps = new Function('return Object.getOwnPropertyNames(this)')();
+    functionProps.forEach((prop: string) => globals.add(prop));
+  } catch (e) {
+    // Ignore if blocked
+  }
+
+  // Remove allowed objects like JSON, Number, Math from the global properties
+  // TODO: do we need to whitelist any other objects?
+  const allowedGlobals = ['JSON', 'Number', 'Math', 'String', 'parseFloat', 'parseInt', 'isNaN', 'console'];
```

Comment:
we will definitely want to check a full list somewhere, but this probably covers the important stuff :+1:

Reply:
done! didn't add absolutely everything, but added some sane defaults using https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference as reference